pkg/failover/server.go

```diff
@@ -45,9 +45,16 @@ func (s *Server) LoggingInterceptor(ctx context.Context, req interface{}, info *
 }
 
 func (s *Server) HealthCheck(ctx context.Context, _ *Empty) (*HealthCheckResponse, error) {
-	return &HealthCheckResponse{
-		Status: HealthCheckResponse_HEALTHY,
-	}, nil
+	var status HealthCheckResponse_Status
+	_, err := s.bouncer.ShowDatabases(ctx)
```

Comment:
How do you feel about returning the error from the ShowDatabases method? Your client would then print out something relevant if PgBouncer was down

Reply:
I think returning an error from this function implies we were unable to complete the RPC, whereas what we're actually trying to communicate is that we successfully determined that the pauser is unhealthy.

Reply:
You could add an `Error` field to the `HealthCheckReponse` struct if you wanted to communicate the error, I guess. But I don't know what you'd do in the future if you're checking multiple components and more than one gives you an error.

Reply:
I think if you return an error you get essentially a 500 back? And that serializes the error, so if you printed the error on the other end it would include the error message

Going to check this though

Reply:
Returning a `errors.New("yolo")` from the health check gives this:

```
λ (lab) ~> go run cmd/stolon-pgbouncer/main.go healthcheck
ts=2019-04-12T10:11:48.050604Z caller=level.go:63 event=healthcheck msg="health checking local pauser API"
ts=2019-04-12T10:11:48.050403Z caller=level.go:63 event=metrics.listen address=127.0.0.1 port=9446
ts=2019-04-12T10:11:48.050868Z caller=level.go:63 event=client.dial client=127.0.0.1
main: error: failed to health check client: rpc error: code = Unknown desc = yolo
```

Reply:
Yeah what I'm saying is might not want your healthcheck RPC to error out if it completed successfully, but one of the components was unhealthy?

I would expect the healthcheck to return successfully (i.e. not 500) and report that the component was unhealthy.

Reply:
Returning an error field or something in the healthcheck struct would probably be the nicest way to verify this, while just returning the error the cheapest to achieve an easy to understand error.

Entirely up to you which we go, but I reckon sending the error message back makes it clear what failed which is nice.

Reply:
> Yeah what I'm saying is might not want your healthcheck RPC to error out if it completed successfully, but one of the components was unhealthy?

I agree with this, I think it's nicer if your health check status is part of the type you return. But am ambivalent as long as we provide the error message.

Reply:
Dunno how you encode this in proto, but what if your health check response looks like this?

```go
type Status int32 // the same proto enum thing
type Component struct {
  Name string
  Status Status
  Error error
}
type HealthCheckResponse struct {
  Status Status
  Components []Component
}
```
The outer `Status` would be the overall status (probably with `Components.all?(healthy?)` semantics), which you could easily check to save effort in the happy path.

I'd like to avoid returning an error when there was no problem carrying out the healthcheck, because it's semantically confusing and just seems like a hack to me.

Reply:
I think ^ is the correct way to do this, would be totally happy to change it to this.